### PR TITLE
String Extension Unit Tests

### PR DESCRIPTION
**Pull request by jwicks31**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

- Unit Tests for String.protoype extensions

Part of #120

Added unit tests for the String.protoype extensions
### Warning

Merging to `master` automatically triggers a deploy in production.

### CODE REVIEW

File: src/extensions/String.test.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import './String'
+
+describe('String.prototype.firstAndLastCharacters', async ( should: any) => {
+  const { assert } = should()
+
+  {
+    const testString = 'testmeonlongstring'
+    assert({
+      given: 'number 6 and string longer than length 12',
+      should: 'return the 6 fist and last characters of the string seperated by ...',
```
Comment (warrenv) [inline src/extensions/String.test.ts]: spelling.

File: src/extensions/String.test.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+import './String'
+
+describe('String.prototype.firstAndLastCharacters', async ( should: any) => {
+  const { assert } = should()
+
+  {
+    const testString = 'testmeonlongstring'
+    assert({
+      given: 'number 6 and string longer than length 12',
```
Comment (warrenv) [inline src/extensions/String.test.ts]: how about something like 'given: "a request for 6 characters...'" to explain what the argument '6' means.

File: src/extensions/String.test.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      expected: 'tthis',
+    })
+  }
+
+  {
+    const testString = 't'
+    
+    assert({
+      given: 'string shorter than supplied targetLength + short padString',
+      should: 'return correct string',
```
Comment (warrenv) [inline src/extensions/String.test.ts]: maybe something like "should: 'return a string with repeated characters from the pad string'".
That would give a good description of the behavior when the pad string is short.

File: src/extensions/String.ts [hunk 4]
```diff
@@ -3,14 +3,15 @@
 interface String {
   firstAndLastCharacters(amount: number): string
   trimLeftByPattern(pattern: string): string
+  padEnd(targetLength: any, padString: any): string
```
Comment (warrenv) [inline src/extensions/String.ts:6]: Does using' number' and 'string' for the argument types cause a problem?

Comment (jwicks31) [reply src/extensions/String.ts:6]: There is logic in the function to handle if the targetLength passed in isn't a number, that's why I didn't use number here. Also it converts padString to a string. So we would have to remove some of the tests if we did add the types.

**Review by warrenv — APPROVED** [review]
* [n/a] Tested changes manually
* [n/a] Checked accidental architectural/style changes
* [x] Reviewed entire diff
* [x] Unit tests
* [n/a] Documentation
* [x] Filenames and locations

**Review by wzalazar — APPROVED** [review]
* [n/a] Tested changes manually
* [n/a] Checked accidental architectural/style changes
* [x] Reviewed entire diff
* [x] Unit tests
* [n/a] Documentation
* [x] Filenames and locations

**Pull request merged**